caikit_nlp/modules/text_generation/peft_config.py

```diff
@@ -96,12 +96,128 @@ def resolve_base_model(base_model, cls, torch_dtype):
 
 
 def get_peft_config(
-    tuning_type, tuning_config, base_model, cls, torch_dtype, verbalizer
+    tuning_type,
+    tuning_config,
+    base_model,
+    cls=None,
+    torch_dtype=None,
+    verbalizer="{{input}}",
 ):
 
+    if isinstance(tuning_type, str):
+        error.value_check(
+            "<NLP65714994E>",
+            tuning_type in TuningType._member_names_,
+            f"Invalid tuning type [{tuning_type}]. Allowed types: "
+            f"[{TuningType._member_names_}]",
+        )
+        tuning_type = TuningType(tuning_type)
+
+    error.type_check("<NLP65714993E>", TuningType, tuning_type=tuning_type)
+
     if tuning_type not in TuningType._member_names_:
         raise NotImplementedError("{} tuning type not supported!".format(tuning_type))
 
+    error.type_check("<NLP65714919E>", PretrainedModelBase, base_model=base_model)
+
+    # Validate if tuned output model type is compatible with base model or not
+    output_model_types = _get_output_types(tuning_config, base_model)
+
+    # NOTE: Base model is a resource at this point
+    task_type = base_model.TASK_TYPE
+
+    error.value_check(
+        "<NLP30542004E>",
+        len(output_model_types) <= base_model.MAX_NUM_TRANSFORMERS,
+        f"Too many output model types. Got {len(output_model_types)}, "
+        f"maximum {base_model.MAX_NUM_TRANSFORMERS}",
+    )
+    # Ensure that our verbalizer is a string and will not render to a hardcoded string
+    error.value_check(
+        "<NLP83837412E>",
```

Comment:
verbalizer is also not used, we are checking if its a valid verbalizer , but that should be done in train of peft_prompt_tuning. it has nothing to do with the Config, so let me know if ok to remove (API change again)

Reply:
I added deprecation warning